### PR TITLE
An Alternative Way to validate slug during edit link

### PR DESCRIPTION
**Pull request by SamuelMwangiW**
A proposed alternative way to validate unique column in links table that uses rules() method instead of property.
Thanks @davidgrzyb for an amazing Youtube series

### CODE REVIEW

File: app/Http/Livewire/EditLink.php [hunk 1]
```diff
@@ -12,11 +12,14 @@ class EditLink extends Component
...
-        'is_enabled' => 'required|boolean',
-    ];
+    public function rules()
+    {
+        return [
+		'url' => 'required|url|max:255',
+		'slug' => 'required|alpha_dash|min:3|max:100|unique:links,slug,'.$this->link_id,
+		'is_enabled' => 'required|boolean',
+	    ];
+    }
```
Comment (davidgrzyb) [inline app/Http/Livewire/EditLink.php]: Can you make sure it is spaced correctly please. Also, we can do the string concatenation like this:
`'slug' => "required|alpha_dash|min:3|max:100|unique:links,slug,{$this->link_id}",`

**Review by davidgrzyb — CHANGES_REQUESTED** [review]
Hey! I appreciate your PR, just need a couple of things fixed :)

File: app/Http/Livewire/EditLink.php (hunk 1 above)
Comment (SamuelMwangiW) [reply app/Http/Livewire/EditLink.php]: Hi @davidgrzyb. Thanks for reviewing my PR. I have updated the string concatenation as requested. If we convert the rules to a property, concatenating with `$this->link_id` throws the error below. 
image

The alternative would be to leave it as a property and add the following line within the `mount()` method : 
```PHP
$this->rules['slug'] = "required|alpha_dash|min:3|max:100|unique:links,slug,{$this->link_id}";
```
I had thought converting the property to a method which allows string concatenations is a more cleaner solution

Comment (SamuelMwangiW) [reply app/Http/Livewire/EditLink.php]: I have pushed a commit with a work around that retains the rules property. This helps maintain the code as written during the video. Kindly review and advise.

**Comment by davidgrzyb** [comment]
Hey, I rewrote the validation myself and committed with you as a co-author [here](https://github.com/davidgrzyb/laravel-url-shortener/commit/ba0fcbb85004b5a7d4923404f8bffa53c8038268). Thanks for the PR, sorry it was easier to just commit the new changes myself.

**Pull request closed without merging**